src/lib/gateway/Gateway.ts

```diff
@@ -0,0 +1,98 @@
+import { RequestHandler, IdKeyed } from '@klasa/request-handler';
+import Collection from '@discordjs/collection';
+import { GatewayStorage } from './GatewayStorage';
+import { Settings } from '../settings/Settings';
+import { Client } from '../types';
+
+export class Gateway extends GatewayStorage {
+
+	/* eslint-disable no-invalid-this */
+	/**
+	 * The cached entries for this Gateway or the external datastore to get the settings from.
+	 */
+	public cache: ProxyMap = (this.name in this.client) && (this.client[this.name as keyof Client] instanceof Map) ?
+		this.client[this.name as keyof Client] as ProxyMap :
+		new Collection<string, ProxyMapEntry>();
+
+	/**
+	 * The request handler that manages the synchronization queue.
+	 */
+	public requestHandler = new RequestHandler(
+		(id: string): Promise<IdKeyed<string>> => {
+			const { provider } = this;
+			return provider === null ?
+				Promise.reject(new Error('Cannot run requests without a provider available.')) :
+				provider.get(this.name, id) as Promise<IdKeyed<string>>;
+		}, (ids: string[]): Promise<IdKeyed<string>[]> => {
+			const { provider } = this;
+			return provider === null ?
+				Promise.reject(new Error('Cannot run requests without a provider available.')) :
+				provider.getAll(this.name, ids) as Promise<IdKeyed<string>[]>;
+		}
+	);
+	/* eslint-enable no-invalid-this */
+
+	/**
+	 * Gets an entry from the cache or creates one if it does not exist
+	 * @param target The target that holds a Settings instance of the holder for the new one
+	 * @param id The settings' identificator
+	 * @example
+	 * // Retrieve a members gateway
```

Comment:
```suggestion
	 * ```
	 * // Retrieve a members gateway
```
Adding backticks make it so the formatting is not lost when hovering over it in VSC. Apply same to all TSDoc.

Reply:
You sure about that?

![image](https://user-images.githubusercontent.com/24852502/70469404-3f3e0800-1ac9-11ea-9727-dff28c5c1033.png)